packages/template-retail-react-app/app/commerce-api/einstein.js

```diff
@@ -53,6 +53,59 @@ class EinsteinAPI {
         return body
     }
 
+    /**
+     * Given a product or item source, returns the product data that Einstein requires
+     */
+    _constructEinsteinProduct(product) {
+        if (product.type && (product.type.master || product.type.variant)) {
+            // handle variants for PDP / viewProduct
+            // Assumes product is a Product object from SCAPI Shopper-Products:
+            // https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-products?meta=type%3AProduct
+            return {
+                id: product.master.masterId,
+                sku: product.id,
```

Comment:
Based on the description, is this supposed to be a variant id first and then master id later for sku?

Reply:
Here we want to set the ID to the master product id and the sku to the variant product id.

With this setup, whenever we're sending an activity related to variants, we are always sending both master and variant ID and the sku will change depending on which variant the user is looking at.